app/lib/katello/repo_discovery.rb

```diff
@@ -42,6 +44,71 @@ def run(resume_point)
 
     private
 
+    def parse_parameter(field_value)
+      field_value.lstrip!
+      i = field_value.index(/[ \t=;,]/) || field_value.length
+      name = field_value.slice!(0, i).downcase(:ascii)
+      field_value.lstrip!
+      if field_value.delete_prefix!('=')
+        field_value.lstrip!
+        if field_value.delete_prefix!('"')
+          value = ''
+          until field_value.empty?
+            break if field_value.delete_prefix!('"')
+            field_value.delete_prefix!("\\")
+            value += field_value.slice!(0, 1) || ''
+          end
+        else
+          i = field_value.index(/[;,]/) || field_value.length
+          value = field_value.slice!(0, i)
+        end
+      end
+      {name: name, value: value || ''}
+    end
+
+    def parse_parameters(field_value)
+      seen_rel = false
+      has_next_rel = false
+      has_anchor = false
+      until field_value.empty?
+        field_value.lstrip!
+        break if field_value.delete_prefix!(';').nil?
+        param = parse_parameter(field_value)
+        case
+        when param[:name] == 'rel' && !seen_rel
+          seen_rel = true
+          has_next_rel = param[:value].downcase(:ascii).split(/[ \t]/).include?('next')
+        when param[:name] == 'anchor'
+          has_anchor = true
+        end
+      end
+      {has_next_rel: has_next_rel, has_anchor: has_anchor}
+    end
+
+    def get_next_link(link_header)
+      # See Appendix B "Algorithms of Parsing Link Header Fiels" of RFC 8288 "Web Linking" (that RFC
```

Comment:
```suggestion
      # See Appendix B "Algorithms of Parsing Link Header Fields" of RFC 8288 "Web Linking" (that RFC
```

Reply:
Think it's worth adding a link to https://www.rfc-editor.org/rfc/rfc8288.html#appendix-B.2 ?